src/devices/machine/tc009xlvc.cpp

```diff
@@ -62,138 +29,102 @@ WRITE8_MEMBER(tc0091lvc_device::vregs_w)
 {
 	if((offset & 0xfc) == 0)
 	{
-		bg0_tilemap->mark_all_dirty();
-		bg1_tilemap->mark_all_dirty();
+		m_bg_tilemap[0]->mark_all_dirty();
+		m_bg_tilemap[1]->mark_all_dirty();
 	}
 
 	m_vregs[offset] = data;
 }
 
-READ8_MEMBER(tc0091lvc_device::tc0091lvc_bitmap_r)
-{
-	return m_bitmap_ram[offset];
-}
-
-WRITE8_MEMBER(tc0091lvc_device::tc0091lvc_bitmap_w)
-{
-	m_bitmap_ram[offset] = data;
-}
-
-
-READ8_MEMBER(tc0091lvc_device::tc0091lvc_pcg1_r)
+WRITE8_MEMBER(tc0091lvc_device::vram_w)
 {
-	return m_pcg1_ram[offset];
-}
+	// TODO : 0x10000-0x13fff Unused?
+	if ((offset & 0xc000) == 0)
+		return;
 
-WRITE8_MEMBER(tc0091lvc_device::tc0091lvc_pcg1_w)
-{
-	m_pcg1_ram[offset] = data;
-	m_gfxdecode->gfx(m_gfx_index)->mark_dirty((offset+0x4000) / 32);
-	tx_tilemap->mark_all_dirty();
-}
+	if((m_vram[offset] & mem_mask) == (data & mem_mask))
```

Comment:
Care to give an explaination of this?
If it's optimization then mem_mask is completely pointless because this is a 8-bit handler.